### ISSUE TITLE
selection and pcadapt output questions

### ISSUE THREAD

**Issue by lkrichardson** [Rosemeis/pcangsd#90]
Hi I have two questions one about --pcadapt and one about --selection.
1. pcadapt- In the paper you describe adjusting the output by the genome inflation factor, how can I do that with my output? I was able to calculate the p-values based on the output with the associated script included in pcangsd. 

2. I also ran --selection without specifying the number of pcs, and got output for 3 pc axes which was what I expected given my admix results. I calculated p-values from the output for each pc axis using your recommendations in R: 
pvals.pc1 <- pchisq(selection[,1], 1, lower.tail=F)
after applying Benjamini–Hochberg, the interesting thing is that none of the pvalues are significant along pc1 or pc3, and only ~22K are significant along pc2. I also looked at the distribution of p-values and they look kind of weird- along pc1 the histogram is hump shaped, and for the other two there is more of a U shape. This is in contrast to the pcadapt output which after Benjamini–Hochberg yielded ~360K significant SNPs from nearly 3 million that were input, and the histogram of pvalues is more uniform especially at the right side of the plot.

I'm wondering why the results between the two methods might be so different and if something strange is going on with one or the other method? Any insights would be helpful, hopefully these questions are appropriate to post here and if not I'm sorry and thanks!

**Comment by Rosemeis** [comment]
Hi,

Thank you for reaching out.
The genomic inflation factor is calculated as follows in R: GIF=median(obs, dof)/qchisq(0.5, dof), where dof is the degrees of freedom.

Could you provide with some more information (sample size, number of SNPs) and maybe output log from pcangsd? :-)
You can also attach the histograms if possible.

Best,
Jonas

**Comment by lkrichardson** [comment]
Hi Jonas, Thanks so much for your response. 
I guess I didn't save the output log unfortunately. Here was my logfile for pcangsd:
PCAngsd v1.11
Time: 19/06/2024 09:26:57
Directory: /home/lkr
Options:
	-beagle /home/lkr/jtgi/angsd/unlinkedSNPs/unlinkedSNPs.beagle.gz
	-out /home/lkr/jtgi/angsd/unlinkedSNPs/pcangsdSelection/pcSelmaf05
	-selection
	-pcadapt
	-sites_save

The input file was beagle output from angsd that was LD-pruned to 6691644 SNPs (at minMaf 0.01).  I didn't change the default for pcangsd, so it filtered down to 2942397 with minMaf 0.05. Here are screenshots of the histograms of pvalues, the pvals.sel.pcX are the histograms of p-values associated with the --selection method, and in contrast you can see the histogram output from pcadapt.

![pvalsHist](https://github.com/Rosemeis/pcangsd/assets/101731315/d9c01622-2b3a-4043-908d-f5dd70348314)

**Comment by Rosemeis** [comment]
Hi,

Thank you, I will have a look!
How are you computing the test statistics component-wise for the z-scores? Using `d2 <- (zscores - median(zscores))^2`?

Best,
Jonas

**Comment by lxsllvn** [comment]
Yes, that's correct. I used D1/D2 to refer to the results for the first/second PC, which I now realize is probably confusing!

Since the K = 1 results from v. 1.36.3 match, I checked how the pcadapt/fastPCA compare to themselves for K = 1 vs. K = 2. 

![Image](https://github.com/user-attachments/assets/5485c910-fc6a-4520-bbb8-f144d45b42ea)
The fastPCA test statistics for PC1 look similar for both K = 1 and K = 2. 

![Image](https://github.com/user-attachments/assets/b84e3bdb-4685-4370-b1ec-11e223997050)
![Image](https://github.com/user-attachments/assets/25564e23-3372-4ff3-8f54-c849d65ef6fb)

But for pcadapt, PC2, rather than PC1, looks like the K =1 PC1 results.

**Comment by Rosemeis** [comment]
You were right! I never flipped the singular matrices in the pcadapt-variant, as I didn't think of the component-wise approach for it. It should be flipped now. Let me know if it produces expected results :-)

Best,
Jonas
